pipeline/metadata/hyperquack_outcome.py

```diff
@@ -0,0 +1,170 @@
+"""Classify outcomes for roundtrip measurements."""
+
+from __future__ import absolute_import
+
+from typing import List, Optional
+
+
+def _status_mismatch(received_status: Optional[str]) -> str:
+  """
+  Args:
+    received_status: status string like "200 OK" or None
+
+  Returns:
+    outcome like "content/status_mismatch" or "content/status_mismatch:404"
+  """
+  if received_status is None or received_status == "":
+    return "content/status_mismatch"
+  return "content/status_mismatch:" + received_status[:3]
+
+
+def _get_first_error(errors: Optional[str]) -> Optional[str]:
+  """
+  Args:
+    errors: string like "error a; error b; error c" or just "error a"
+
+  Feturns
+    "error a"
+  """
+  if errors is None:
+    return None
+
+  if ';' in errors:
+    return errors[:errors.index(';')]
+  return errors  # single error
+
+
+def _has_akamai_server_header(received_headers: List[str]) -> bool:
+  for received_header in received_headers:
+    if received_header in ('Server: AkamaiGHost', 'Server: GHost'):
+      return True
+  return False
+
+
+def _classify_hyperquack_error(error: Optional[str], scan_type: str,
+                               received_status: Optional[str]) -> str:
+  """
+  Args:
+    error string like 'read tcp 141.212.123.235:11397->117.78.42.54:9: read: connection reset by peer'
+    scan type: like 'echo, 'discard', etc
+
+  Returns:
+    outcome: string like "<stage>/<result>"
+  """
+  # yapf: disable
+  # pylint: disable=multiple-statements
+  # pylint: disable=too-many-return-statements
+  # pylint: disable=too-many-branches
+  # pylint: disable=too-many-statements
+
+  # Success
+  if error is None or error == "": return "expected/match"
+
+  # System failures
```

Comment:
Move this to a `is_system_failure` method, and do:
```py
if (is_system_failure(error)): return "setup/system_failure"
```

Move it to classify_hyperquack_outcome

Reply:
done